beacon-chain/blockchain/optimistic_sync.go

```diff
@@ -37,17 +36,16 @@ func (s *Service) optimisticCandidateBlock(ctx context.Context, blk block.Beacon
 // loadSyncedTips loads a previously saved synced Tips from DB
 // if no synced tips are saved, then it creates one from the given
 // root and slot number.
-func (s *Service) loadSyncedTips(root [32]byte, slot types.Slot) error {
+func (s *Service) loadSyncedTips(root [32]byte, slot types.Slot) {
 	// Initialize synced tips
 	tips, err := s.cfg.BeaconDB.ValidatedTips(s.ctx)
-	if err != nil {
-		return errors.Wrap(err, "could not get synced tips")
-	}
-	if len(tips) == 0 {
+	if err != nil || len(tips) == 0 {
 		tips[root] = slot
+		if err != nil {
+			log.WithError(err).Error("could not read synced tips from DB")
+		}
 	}
 	if err := s.cfg.ForkChoiceStore.SetSyncedTips(tips); err != nil {
-		return errors.Wrap(err, "could not set synced tips")
+		log.WithError(err).Error("could not save synced tips to fork choice")
```

Comment:
I think the success of set synced tips to fork choice store is necessary for the node to function. Returning an error is still best here. I much rather be conservative and fail fast